test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+// ============================================================================
+void fastq_write(benchmark::State & state)
+{
+    std::ostringstream ostream;
+
+    sequence_file_format_fastq format;
+    sequence_file_output_options options{};
+
+    std::string const id{"@name"};
+    auto seq = test::generate_sequence<dna5>(SEQUENCE_LENGTH, 0, SEED);
+    auto quality = test::generate_sequence<phred42>(SEQUENCE_LENGTH, 0, SEED);
+
+    for (auto _ : state)
+    {
+        format.write(ostream, options, seq, id, quality);
+    }
+}
+
+// ============================================================================
+// read dummy fastq file from a stream
+// ============================================================================
+template<std::size_t n_entries_in_file = N_ENTRIES_IN_FILE>
+void fastq_read_from_stream(benchmark::State & state)
+{
+    sequence_file_format_fastq format;
+    sequence_file_input_options<dna5, false> const options{};
+
+    std::string id{};
+    std::vector<dna5> seq{};
+    std::vector<phred42> qual{};
+
+    for (auto _ : state)
+    {
+        std::istringstream istream{get_file(n_entries_in_file)};
+        // refilling streams introduces the same constant to
+        // both seqan3 and 2 benchmark runtime so they are still comparable
+
+        format.read(istream, options, seq, id, qual);
+
+        id.clear();
+        seq.clear();
+        qual.clear();
+    }
+}
+
+// ============================================================================
+// seqan2 comparison
+// ============================================================================
+#if SEQAN2_HAS_SEQAN3
+
+template<std::size_t n_entries_in_file = N_ENTRIES_IN_FILE>
+void fastq_read_from_stream_seqan2(benchmark::State & state)
+{
+    using namespace seqan;
+
+    std::istringstream istream{};
+
+    auto restart_iterator = [&istream]()    // c.f. format_fasta_benchmark
+    {
+        istream = std::istringstream{get_file(n_entries_in_file)};
+        // same constant as seqan3 benchmark
+
+        seqan::VirtualStream<char, Input> comp{};
+        open(comp, istream);
+        return directionIterator(comp, Input());
+    };
+
+    String<char> id{};
+    String<Dna5> seq{};
+    String<char> qual{};
+
+    for (auto _ : state)
+    {
+        auto it = restart_iterator();
+        readRecord(id, seq, qual, it, seqan::Fastq{});
+
+        clear(id);
+        clear(seq);
+        clear(qual);
+    }
+}
+
+#endif
+
+// ============================================================================
+// read dummy fastq file from temporary file on disk
+// ============================================================================
+template<std::size_t n_entries_in_file = N_ENTRIES_IN_FILE>
+void fastq_read_from_disk(benchmark::State & state)
+{
+    sequence_file_format_fastq format;
+
+    // create temporary file on disk
+    std::filesystem::path tmp_dir = std::filesystem::temp_directory_path();
```

Comment:
in `#include <seqan3/test/tmp_filename.hpp>` we have a `tmp_filename filename{"cereal_test"};` that you can use which creates a file in the tempdirectory for you and even removes it afterwards.

Reply:
fixed